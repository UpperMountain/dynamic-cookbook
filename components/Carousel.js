import React from "react";
import { ScrollView } from "react-native";
import Padded from "./Padded";

const Carousel = ({ children }) => (
  <ScrollView horizontal style={{ overflow: "visible" }}>
    <Padded horizontal style={{ display: "flex", flexDirection: "row" }}>
      {children}
    </Padded>
  </ScrollView>
);

export default Carousel;
