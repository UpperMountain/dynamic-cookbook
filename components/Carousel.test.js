import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";
import Carousel from "./Carousel";

it("should render something without exploding", () => {
  renderer.create(<Carousel />);
});

it("should render a list with two items", () => {
  const cmp = renderer.create(
    <Carousel>
      {/* don't actually render meal cards, so we only snapshot this component */}
      <Text>Test1</Text>
      <Text>Test2</Text>
    </Carousel>
  );

  expect(cmp.toJSON()).toMatchSnapshot();
});
