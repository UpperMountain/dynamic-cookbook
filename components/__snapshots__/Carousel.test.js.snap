// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list with two items 1`] = `null`;
